### PR TITLE
Use PXF server from apache/hawq to the new greenplum/pxf repo

### PR DESCRIPTION
**Pull request by shivzone**
PXF client in gpdb uses pxf libraries from apache hawq repo. These pxf libraries will continue being developed in a new PXF repo https://github.com/greenplum-db/pxf and is in the process of getting open sourced in the next few days. The PXF extension and gpdb-pxf client code will continue to remain in gpdb repo.

The following changes are included in this PR:
1. Transition from the old PXF namespace org.apache.hawq.pxf to org.greenplum.pxf
(there is a separate PR in the PXF repo to address the package namespace refactor
https://github.com/greenplum-db/pxf/pull/5)

2. Doc updates to reflect the new PXF repo and the new package namespace

### CODE REVIEW

**Comment by ivannovick** [comment]
congrats on this milestone

File: gpdb-doc/markdown/pxf/sdk/dev_overview.html.md.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Using the PXF Java SDK
 ---
 The Greenplum Platform Extension Framework (PXF) SDK provides the Java classes and interfaces that you use to create connectors to new external data sources, data formats, and data access APIs from Greenplum Database. You can extend PXF functionality *without changing Greenplum Database* when you use the PXF Java SDK.
 
-PXF in Greenplum Database is based on [PXF](https://cwiki.apache.org/confluence/display/HAWQ/PXF) from the open source Apache HAWQ (incubating) project. You can contribute to PXF development via the [Greenplum Database](https://github.com/greenplum-db/gpdb/tree/master/gpAux/extensions/pxf) and the [Apache HAWQ (incubating)](https://github.com/apache/incubator-hawq/tree/master/pxf) open source `github` repositories. 
+PXF in Greenplum Database has its roots from Apache HAWQ project. You can contribute to PXF development via [PXF](https://github.com/greenplum-db/pxf) and the [Greenplum Database PXF Extension(Client)](https://github.com/greenplum-db/gpdb/tree/master/gpAux/extensions/pxf) and the  open source `github` repositories.
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/sdk/dev_overview.html.md.erb]: what about "PXF in Greenplum Database has its roots **in the** Apache HAWQ project.  You can can contribute to **Greenplum** PXF development via **the** open source `github` repositories for PXF (server) and the Greenplum Database PXF Extension (client).  (keep the links).

File: gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb [hunk 2]
```diff
@@ -111,14 +111,14 @@ Before building the *Demo* connector, ensure that you have:
 
 Perform the following procedure to create a local copy of the *Demo* connector source code, update package names, configure compile-time dependencies, and use `gradle` to build the connector.
 
-1. Download the PXF *Demo* connector source code. You can obtain the PXF source code from the Apache HAWQ (incubating) `incubator-hawq` `github` repository. For example:
+1. Download the PXF *Demo* connector source code. You can obtain the PXF source code from PXF `github` repository. For example:
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: ... source code from **the Greenplum** PXF github ...

File: gpdb-doc/markdown/pxf/overview_pxf.html.md.erb [hunk 3]
```diff
@@ -21,7 +21,7 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-The Greenplum Platform Extension Framework (PXF) provides parallel, high throughput data access and federated queries across heterogeneous data sources via built-in connectors that map a Greenplum Database external table definition to an external data source. This Greenplum Database extension is based on [PXF](https://cwiki.apache.org/confluence/display/HAWQ/PXF) from Apache HAWQ (incubating).
+The Greenplum Platform Extension Framework (PXF) provides parallel, high throughput data access and federated queries across heterogeneous data sources via built-in connectors that map a Greenplum Database external table definition to an external data source. This Greenplum Database extension [PXF](https://github.com/greenplum-db/pxf) has its roots from Apache HAWQ.
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/overview_pxf.html.md.erb]: this page is in the pxf end user docs.  i don't think we need to include a link to the pxf repository.  what about:

PXF in Greenplum Database has its roots in the Apache HAWQ project.

File: gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb [hunk 4]
```diff
@@ -134,18 +134,18 @@ Perform the following procedure to create a local copy of the *Demo* connector s
     user@devsystem$ cp $PXFDEV_BASE/pxf-api-<version>.jar libs/
     ```
     
-4. The source code for the PXF *Demo* connector is located in the `incubator-hawq/pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/examples` directory of the repository you cloned in Step 1. Copy this code to your work area. For example:
+4. The source code for the PXF *Demo* connector is located in the `pxf/pxf/pxf-api/src/main/java/org/greenplum/pxf/api/examples` directory of the repository you cloned in Step 1. Copy this code to your work area. For example:
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: when i first saw this, i thought the pxf/pxf/ in the file path was a typo.  then i realized that the repository name is pxf, and that there is a pxf/ directory at the top level of the repository.  (this directory contains the pxf source code.)

i am not necessarily advocating for a new name for this directory, just pointing this out.

**Comment by lisakowen** [comment]
@danielgustafsson - here is my understanding; @shivram please confirm:  pxf has a client and a server component.  the pxf client in greenplum is the C extension; this will remain in the greenplum-db/gpdb repository.  the pxf java server component is from hawq.  the team is in the process of creating a new "pxf server for greenplum" repository for pxf greenplum working going forward.  not sure on the timing of the repository.  this PR addresses some changes related to changing the pxf server package name for this effort.

**Comment by shivzone** [comment]
@lisakowen thanks for making it clear. I have updated the PR description to include more context for this change.
@danielgustafsson yes we generally use forks, the original committer of the branch wasn't aware of the practice, and we'll make sure this doesn't happen. Can we retain this branch for now until the PR is merged to retain the comments exchanged so far ?

File: src/tools/docker/centos6-gpadmin/Dockerfile [hunk 5]
```diff
@@ -19,3 +19,4 @@ RUN groupadd -g 1000 gpadmin && useradd -u 1000 -g 1000 gpadmin && \
     chown -R gpadmin:gpadmin /usr/local/greenplum-db-devel
 
 USER gpadmin
+
```
Comment (lavjain) [inline src/tools/docker/centos6-gpadmin/Dockerfile]: This file can be removed from the PR since it was not modified.

File: README.md [hunk 6]
```diff
@@ -171,6 +171,7 @@ make distclean
 
 PXF is an extension framework for GPDB to enable fast access to external hadoop datasets.
 Refer to [PXF extension](https://github.com/greenplum-db/gpdb/tree/master/gpAux/extensions/pxf) for more information.
```
Comment (danielgustafsson) [inline README.md]: This link should be against the actual README and not an external link into the code:
```markdown
[PXF extension](gpAux/extensions/pxf/README.md)
```
While not introduced in this PR, this is a good time to fix it.

File: gpAux/extensions/pxf/README.md [hunk 7]
```diff
@@ -54,23 +45,26 @@ Additional instructions on building and starting a GPDB cluster can be
 found in the top-level [README.md](../../../README.md) ("_Build the
 database_" section).
 
-## Create and use PXF external table
-If you wish to simply test GPDB and PXF without hadoop, you can use the Demo Profile.
+
+### Install PXF Server
+Please refer to [PXF Development](https://github.com/greenplum-db/pxf/blob/master/README.md) for instructions to setup PXF.
+Once you install and run PXF server alongside the GPDB segments, you can select data from the demo PXF profile:
```
Comment (danielgustafsson) [inline gpAux/extensions/pxf/README.md]: Trailing colon that doesn't lead to any content. Should be a period instead?

File: gpAux/extensions/pxf/README.md [hunk 8]
```diff
@@ -54,23 +45,26 @@ Additional instructions on building and starting a GPDB cluster can be
...
+
+### Create and use PXF external table
+If you wish to simply test drive PXF extension without hadoop, you can avoid installing the hadoop components and simply use the Demo Profile.
 
 The Demo profile demonstrates how GPDB can parallely the external data via the PXF agents. The data served is
```
Comment (danielgustafsson) [inline gpAux/extensions/pxf/README.md]: This sentence seems incomplete, can parallely **what** the external data? Also, parallely is incorrectly spelled, should be "parallelly".

Again, not introduced in this PR but a documentation PR is a good place to fix documentation.

File: gpAux/extensions/pxf/README.md (hunk 7 above)
Comment (danielgustafsson) [reply gpAux/extensions/pxf/README.md]: This is due to the text being cut/pasted from a section further down where the content still exists.

File: gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb [hunk 9]
```diff
@@ -134,18 +134,18 @@ Perform the following procedure to create a local copy of the *Demo* connector s
...
 
-5. The original PXF *Demo* connector resides in the `org.apache.hawq.pxf.api.examples` package. Your *Demo* connector resides in a package named `org.greenplum.pxf.example.demo`. Update the package name in your local copy of the *Demo* connector source code. You can edit the files, run a script, etc. For example:
+5. The original PXF *Demo* connector resides in the `org.greenplum.pxf.api.examples` package. Your *Demo* connector resides in a package named `org.greenplum.pxf.example.demo`. Update the package name in your local copy of the *Demo* connector source code. You can edit the files, run a script, etc. For example:
 
     ``` shell
-    user@devsystem$ sed -i.bak s/"org.apache.hawq.pxf.api.examples"/"org.greenplum.pxf.example.demo"/g *.java
+    user@devsystem$ sed -i.bak s/"org.greenplum.pxf.api.examples"/"org.greenplum.pxf.example.demo"/g *.java
     ```
 
     The `sed` command above creates a backup of each file. Remove the backup files. For example:
```
Comment (danielgustafsson) [inline gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: Why do we ask the user to create backups and then immediately remove them?

Comment (lisakowen) [reply gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: @danielgustafsson - this was the only sed command that i could come up with at the time that worked on both macos and centos.  if you have a better functionally-equivalent command, please pass it along.  thanks!

**Comment by shivzone** [comment]
@danielgustafsson I've addressed your suggestions

File: README.md [hunk 10]
```diff
@@ -170,7 +170,7 @@ make distclean
 ### Building GPDB with PXF
 
 PXF is an extension framework for GPDB to enable fast access to external hadoop datasets.
-Refer to [PXF extension](https://github.com/greenplum-db/gpdb/tree/master/gpAux/extensions/pxf) for more information.
+Refer to [PXF extension](https://github.com/greenplum-db/gpdb/blob/master/gpAux/extensions/pxf/README.md) for more information.
```
Comment (danielgustafsson) [inline README.md]: When the target file is in the same repo there is no need to invoke a http url, just point to the file on the filesystem like `[PXF extension](gpAux/extensions/pxf/README.md)`

File: gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb [hunk 11]
```diff
@@ -134,21 +134,21 @@ Perform the following procedure to create a local copy of the *Demo* connector s
...
-5. The original PXF *Demo* connector resides in the `org.apache.hawq.pxf.api.examples` package. Your *Demo* connector resides in a package named `org.greenplum.pxf.example.demo`. Update the package name in your local copy of the *Demo* connector source code. You can edit the files, run a script, etc. For example:
+5. The original PXF *Demo* connector resides in the `org.greenplum.pxf.api.examples` package. Your *Demo* connector resides in a package named `org.greenplum.pxf.example.demo`. Update the package name in your local copy of the *Demo* connector source code. You can edit the files, run a script, etc. For example:
 
     ``` shell
-    user@devsystem$ sed -i.bak s/"org.apache.hawq.pxf.api.examples"/"org.greenplum.pxf.example.demo"/g *.java
+    user@devsystem$ sed -i.bak s/"org.greenplum.pxf.api.examples"/"org.greenplum.pxf.example.demo"/g *.java
     ```
 
-    The `sed` command above creates a backup of each file. Remove the backup files. For example:
+    Remove the temporary files created by the previous command:
```
Comment (danielgustafsson) [inline gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: If we want the user to remove backups immediately, we can just do `sed -i s/"org..` without extension and avoid the backups in the first place. Either that, or reword to describe what the point of the backups is, something like:  _"Backup files will be created by the previous command. When the update has been verified to be successful, the backups can be removed with the following command:"_

**Comment by danielgustafsson** [comment]
> On 20 Sep 2018, at 00:30, Shivram Mani <notifications@github.com> wrote:

> @danielgustafsson yes we generally use forks, the original committer of the branch wasn't aware of the practice, and we'll make sure this doesn't happen. Can we retain this branch for now until the PR is merged to retain the comments exchanged so far ?

Of course, it was more for future reference, let’s keep this PR as it is now when it’s here.

File: gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb (hunk 11 above)
Comment (shivzone) [reply gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: @lisakowen sed -i is generally flaky on mac. sed -i -e fixes this issue without having to create backups

Comment (shivzone) [reply gpdb-doc/markdown/pxf/sdk/build_conn.html.md.erb]: @lisakowen sed takes the argument after -i as the backup extensions. Using -i '' would fix it

**Pull request merged**